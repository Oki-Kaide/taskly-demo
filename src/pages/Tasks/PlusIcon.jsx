import React from 'react';

const PlusIcon = () => (
<svg width="14px" height="14px" viewBox="0 0 14 14" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlnsXlink="http://www.w3.org/1999/xlink">
<g id="Page-1" stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
    <g id="Website-05" transform="translate(-1091.000000, -428.000000)">
        <g id="Group-3" transform="translate(760.000000, 176.000000)">
            <g id="add-24px-(1)" transform="translate(326.000000, 247.000000)">
                <polygon id="Path" fill="#999999" fillRule="nonzero" points="19 13 13 13 13 19 11 19 11 13 5 13 5 11 11 11 11 5 13 5 13 11 19 11"></polygon>
                <polygon id="Path" points="0 0 24 0 24 24 0 24"></polygon>
            </g>
        </g>
    </g>
</g>
</svg>
);

export default PlusIcon;
