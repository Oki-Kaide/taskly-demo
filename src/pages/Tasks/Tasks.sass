@import '../../styles/variables.sass'

.task-item
  outline: 0
  border: none
  border-bottom: 1px solid $taskly-red
  height: 30px
  line-height: 21px
  font-size: 14px
  &::placeholder
    font-family: CircularStd
  &:-ms-input-placeholder
    font-family: CircularStd
  &::-ms-input-placeholder
    font-family: CircularStd
.add-task
  display: flex
  flex-direction: row
  justify-content: space-between
  margin-top: 22px
  cursor: pointer
.finish-button
  display: flex
  flex-direction: row
  justify-content: center
  width: 100%
  margin-top: 50px