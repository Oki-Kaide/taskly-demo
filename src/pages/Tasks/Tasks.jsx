import React from 'react';
import { LoggedInLayout, RedButton } from '../../components';
import PlusIcon from './PlusIcon';
import './Tasks.sass';

const Tasks = ({accountData, logout}) => {
  return (
    <LoggedInLayout mainImageUrl="./images/smiling-man-bust.png" avatar={accountData.avatar} logout={logout}>
      <h1>
        Thanks, now let's get started!
      </h1>
      <div className="description">
        What can we help you with? Add the tasks below that you would like help with.
      </div>
      <input className="task-item" placeholder="New Task"></input>
      <div className="description add-task">
        <span>Add another task</span>
        <PlusIcon />
      </div>
      <RedButton text="Finish" className="finish-button" />
    </LoggedInLayout>
  );
}

export default Tasks;