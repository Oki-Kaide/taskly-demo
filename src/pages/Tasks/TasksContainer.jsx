import React from 'react';
import PropTypes from 'prop-types';
import Tasks from './Tasks';

class TasksContainer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    const { accountData, logout } = this.props;
    return(
      <Tasks accountData={accountData} logout={logout} />
    );
  }
}

TasksContainer.propTypes = {
  logout: PropTypes.func.isRequired,
  accountData: PropTypes.shape().isRequired,
}
export default TasksContainer;