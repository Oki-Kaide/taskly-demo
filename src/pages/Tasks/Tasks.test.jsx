import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Tasks from './Tasks';
import PlusIcon from './PlusIcon';

const createTestProps = props => ({
  accountData: {
    name: 'test',
    avatar: '',
  },
  logout: jest.fn(),
  ...props,
});

describe('Tasks component', () => {
  let props;

  beforeEach(() => {
    props = createTestProps({});
  });

  it('renders without crashing', () => {
    shallow(<Tasks {...props} />);
  });

  it('matches the rendered snapshot', () => {
    const component = shallow(<Tasks {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});

describe('PlusIcon component', () => {
  it('renders without crashing', () => {
    shallow(<PlusIcon />);
  });

  it('matches the rendered snapshot', () => {
    const component = shallow(<PlusIcon />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
