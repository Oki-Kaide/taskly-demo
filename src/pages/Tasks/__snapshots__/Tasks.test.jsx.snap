// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlusIcon component matches the rendered snapshot 1`] = `
<svg
  height="14px"
  version="1.1"
  viewBox="0 0 14 14"
  width="14px"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <g
    fill="none"
    fillRule="evenodd"
    id="Page-1"
    stroke="none"
    strokeWidth="1"
  >
    <g
      id="Website-05"
      transform="translate(-1091.000000, -428.000000)"
    >
      <g
        id="Group-3"
        transform="translate(760.000000, 176.000000)"
      >
        <g
          id="add-24px-(1)"
          transform="translate(326.000000, 247.000000)"
        >
          <polygon
            fill="#999999"
            fillRule="nonzero"
            id="Path"
            points="19 13 13 13 13 19 11 19 11 13 5 13 5 11 11 11 11 5 13 5 13 11 19 11"
          />
          <polygon
            id="Path"
            points="0 0 24 0 24 24 0 24"
          />
        </g>
      </g>
    </g>
  </g>
</svg>
`;

exports[`Tasks component matches the rendered snapshot 1`] = `
<LoggedInLayout
  avatar=""
  logout={[MockFunction]}
  mainImageUrl="./images/smiling-man-bust.png"
  openConfirmModal={null}
>
  <h1>
    Thanks, now let's get started!
  </h1>
  <div
    className="description"
  >
    What can we help you with? Add the tasks below that you would like help with.
  </div>
  <input
    className="task-item"
    placeholder="New Task"
  />
  <div
    className="description add-task"
  >
    <span>
      Add another task
    </span>
    <PlusIcon />
  </div>
  <RedButton
    className="finish-button"
    handleOnClick={[Function]}
    isLoading={false}
    text="Finish"
  />
</LoggedInLayout>
`;
